external_search.py

```diff
@@ -55,27 +55,32 @@ def __init__(self, _db, url=None, works_index=None):
         self.works_alias = None
         integration = None
 
-        if not ExternalSearchIndex.__client:
-            if _db and (not url or not works_index):
-                integration = ExternalIntegration.lookup(
-                    _db, ExternalIntegration.ELASTICSEARCH,
-                    goal=ExternalIntegration.SEARCH_GOAL
-                )
+        if not _db:
+            raise CannotLoadConfiguration(
+                "Cannot load Elasticsearch configuration without a database.",
+            )
+        if not url or not works_index:
+            integration = ExternalIntegration.lookup(
+                _db, ExternalIntegration.ELASTICSEARCH,
+                goal=ExternalIntegration.SEARCH_GOAL
+            )
 
-                if not integration:
-                    raise CannotLoadConfiguration(
-                        "No Elasticsearch integration configured."
-                    )
-                url = url or integration.url
-                if not works_index:
-                    setting = integration.setting(self.WORKS_INDEX_KEY)
-                    works_index = setting.value_or_default(
-                        self.DEFAULT_WORKS_INDEX
-                    )
-            if not url:
+            if not integration:
                 raise CannotLoadConfiguration(
-                    "No URL configured to Elasticsearch server."
+                    "No Elasticsearch integration configured."
+                )
+            url = url or integration.url
+            if not works_index:
+                setting = integration.setting(self.WORKS_INDEX_KEY)
+                works_index = setting.value_or_default(
+                    self.DEFAULT_WORKS_INDEX
```

Comment:
Before, if __client already existed, the integration and settings weren't looked up again. But the works_index was still being used to set the index and alias. Now, the integration is looked up every time, but the client is still only initialized once.